scheduler/src/cook/compute_cluster.clj

```diff
@@ -28,10 +28,14 @@
   (db-id [this]
     "Get a database entity-id for this compute cluster (used for putting it into a task structure).")
 
+  (initialize-cluster [this pool->fenzo pool->offers-chan]
+    "Initializes the cluster. Returns a channel that will be delivered on when the cluster loses leadership.")
```

Comment:
Should we worry about when the cluster regains leadership? Or do we expect the cluster to exit when it loses leadership? Either way, should we clarify in the docstring?

Reply:
We should clarify what the channel values can be, e.g. an exception is supported as a return value and `nil` when there was no error.

Reply:
We don't expect to regain leadership, since we'll keep the existing pattern of giving up cook leadership if we lose compute cluster leadership.

Added both to the docstring.